documentation/modules/oauth/proc-oauth-client-config.adoc

```diff
@@ -47,8 +58,49 @@ System.setProperty(ClientConfig.OAUTH_AUDIENCE, "_<audience_value_") <5>
 An authorization server may require a client to specify the scope.
 <5> (Optional) The `audience` for requesting the token from the token endpoint.
 An authorization server may require a client to specify the audience.
+--
++
+--
+.Password grants mechanism properties 
+[source,env, subs="+quotes,attributes"]
+----
+System.setProperty(ClientConfig.OAUTH_TOKEN_ENDPOINT_URI, “_https://<auth-server-address>/auth/realms/master/protocol/openid-connect/token_”);
+System.setProperty(ClientConfig.OAUTH_CLIENT_ID, "_<client_name>_"); <1>
+System.setProperty(ClientConfig.OAUTH_CLIENT_SECRET, "_<client_secret>_"); <2>
+System.setProperty(ClientConfig.OAUTH_PASSWORD_GRANT_USERNAME, "__<username>__"); <3>
+System.setProperty(ClientConfig.OAUTH_PASSWORD_GRANT_PASSWORD, "__<password>__"); <4>
```

Comment:
I find this way a bit weird. Why would someone really use this way to set up the properties? I guess the more logical configuration options are to use environment variables, pass system properties from outside using `-D` option or configure it directly in the `sasl.jaas.config` option in the Kafka configuration. E.g.

```
sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
  oauth.client.id="team-a-client" \
  oauth.client.secret="team-a-client-secret" \
  ...
  ...;
```

The mechanism to use system properties but configured from inside the application should technically work, but does seem like something one would use in production-grade applications. This applies to this whole section basically.

Reply:
Like we do in the later examples. @mstruk -- should we change how we add the config here?  We might also want to mention the various options.

Reply:
Yes, I guess let's make it the way we ourselves configure it as @scholzj suggested.

I'm guessing that we chose this style because it's less error prone. Setting `sasl.jaas.config` requires following a strict formatting, it's supposed to be a 'single line' (which `\n` at the end of each line ensures), the config parameter names _must not_ be quoted, but all the values _have to be_ quoted. It has to start exactly like it does, contain the keyword `required`, and it has to end with `;`.


In this case, for client credentials it would be:
```
sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
  oauth.token.endpoint.uri="_<token_endpoint_url>_" \
  oauth.client.id="_<client_id>_"  <1> \
  oauth.client.secret="_<client_secret>_" <2> ;
```

For 'password grant' we would have:
```
sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
  oauth.token.endpoint.uri="_<token_endpoint_url>_" \
  oauth.client.id="_<client_id>_"  <1> \
  oauth.client.secret="_<client_secret>_" <2> \
  oauth.password.grant.username="__<username>__"  <3> \
  oauth.password.grant.password="__<password>__"  <4> ;
```

For access token we would have:
```
sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
  oauth.token.endpoint.uri="_<token_endpoint_url>_" \
  oauth.access.token="_<access_token>_"  <1> ;
```

And for refresh token it would be:
```
sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
  oauth.token.endpoint.uri="_<token_endpoint_url>_" \
  oauth.client.id="_<client_id>_"  <1> \
  oauth.client.secret="_<client_secret>_" <2> \
  oauth.refresh.token="_<refresh_token>_"  <3> ;
```